bootstrap/system_dependency.py

```diff
@@ -0,0 +1,168 @@
+    replace_dict = {"lua": "DEVCOM.Lua",
+                    "python": "python",
+                    "patch": "",
+                    "bison": "",
+                    "flex": ""}
+    command = "winget install --disable-interactivity --accept-package-agreements {}".format(
+        _replace_wholewords(dependencies_str, replace_dict))
+    assert _run_command_with_confirm(command, no_confirm)
+
+
+def _install(dependencies_str: str, no_confirm: bool):
+    platform_system = platform.system()
+    if platform_system == "Darwin":
+        _install_with_brew(dependencies_str, no_confirm)
+    elif platform_system == "Linux":
+        distro_id = distro.id()
+        if distro_id == "ubuntu":
+            _install_with_apt(dependencies_str, no_confirm)
+        elif "arch" in distro_id or "manjaro" in distro_id:
+            _install_with_pacman(dependencies_str, no_confirm)
+        elif "rocky" in distro_id:
+            _install_with_dnf(dependencies_str, no_confirm)
+        else:
+            sys.exit(f"Unsupported platform {distro_id} exiting")
+    elif platform_system == "Windows":
+        _install_with_winget(dependencies_str, no_confirm)
+    else:
+        sys.exit(f"Unsupported platform {platform_system} exiting")
+
+
+def _create_system_dependencies(minifi_options: MinifiOptions) -> str:
+    system_dependencies = {'patch': 'patch', 'make': 'make'}
+    if minifi_options.is_enabled("ENABLE_EXPRESSION_LANGUAGE"):
+        system_dependencies['bison'] = 'bison'
+        system_dependencies['flex'] = 'flex'
+    if minifi_options.is_enabled("ENABLE_LIBARCHIVE"):
+        system_dependencies['libarchive'] = 'libarchive'
+    if minifi_options.is_enabled("ENABLE_PCAP"):
+        system_dependencies['libpcap'] = 'libpcap'
+    if minifi_options.is_enabled("ENABLE_USB_CAMERA"):
+        system_dependencies['libusb'] = 'libusb'
+        system_dependencies['libpng'] = 'libpng'
+    if minifi_options.is_enabled("ENABLE_GPS"):
+        system_dependencies['gpsd'] = 'gpsd'
+    if minifi_options.is_enabled("ENABLE_COAP"):
+        system_dependencies['automake'] = 'automake'
+        system_dependencies['autoconf'] = 'autoconf'
+        system_dependencies['libtool'] = 'libtool'
+    if minifi_options.is_enabled("ENABLE_LUA_SCRIPTING"):
+        system_dependencies['lua'] = 'lua'
+    if minifi_options.is_enabled("ENABLE_PYTHON_SCRIPTING"):
+        system_dependencies['python'] = 'python'
+    if minifi_options.is_enabled("MINIFI_OPENSSL"):
+        system_dependencies['openssl'] = 'perl'
+    if minifi_options.is_enabled("ENABLE_JNI"):
+        system_dependencies['jni'] = ''
+    return " ".join(str(value) for value in system_dependencies.values())
+
+
+def install_required(minifi_options: MinifiOptions):
+    _install(_create_system_dependencies(minifi_options), minifi_options.no_confirm)
+
+
+def install_compiler(no_confirm: bool) -> str:
+    print("For CMake to work we need a working compiler")
```

Comment:
This should check first whether a compiler is already available on the system.

Reply:
Now it only installs these packages if they are missing. You can also skip the installation of the default compiler with a flag aswell